rules/manager.go

```diff
@@ -907,6 +907,34 @@ func (m *Manager) AlertingRules() []*AlertingRule {
 	return alerts
 }
 
+// AlertingRules returns the list of the manager's alerting rules organised by group.
+func (m *Manager) AlertingRulesbyGroup() *AlertingGroups {
+	m.mtx.RLock()
+	defer m.mtx.RUnlock()
+
+	alerts := &AlertingGroups{}
+	for _, g := range m.groups {
+		group := &AlertingGroup{Name: g.name}
+		for _, rule := range g.rules {
+			if alertingRule, ok := rule.(*AlertingRule); ok {
+				group.Rules = append(group.Rules, alertingRule)
+			}
+		}
+		alerts.Groups = append(alerts.Groups, group)
+	}
+	return alerts
+}
+
+type AlertingGroups struct {
```

Comment:
I'm not sure we need this intermediary type. Just return a slice of `[]*rule.Group` directly from the method that returns the rule groups?